astropy/cosmology/core.py

```diff
@@ -93,8 +97,12 @@ class Cosmology(metaclass=abc.ABCMeta):
     write = UnifiedReadWriteMethod(CosmologyWrite)
 
     # Parameters
-    __parameters__: tuple[str, ...] = ()
-    __all_parameters__: tuple[str, ...] = ()
+    parameters = ParameterDescriptor()
+    derived_parameters = DerivedParameterDescriptor()
+
+    _parameters_: ClassVar = MappingProxyType[str, Parameter]({})
```

Comment:
Why the underscore on the trailing side too?

Reply:
I can drop it. It's mimicking python dunder properties, but not necessary.

Reply:
Done.